selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/dashboard/workspaces/WorkspacesListTest.java

```diff
@@ -0,0 +1,380 @@
+    assertTrue(workspaces.isWorkspaceChecked(blankWorkspaceName));
+    assertTrue(!workspaces.isBulkCheckboxEnabled());
+    workspaces.waitDeleteWorkspaceBtn();
+
+    workspaces.selectWorkspaceByCheckbox(javaWorkspaceName);
+
+    assertTrue(workspaces.isWorkspaceChecked(javaWorkspaceName));
+    assertTrue(workspaces.isWorkspaceChecked(blankWorkspaceName));
+    assertTrue(workspaces.isBulkCheckboxEnabled());
+    workspaces.waitDeleteWorkspaceBtn();
+
+    workspaces.selectAllWorkspacesByBulk();
+
+    assertTrue(!workspaces.isWorkspaceChecked(javaWorkspaceName));
+    assertTrue(!workspaces.isWorkspaceChecked(blankWorkspaceName));
+    assertTrue(!workspaces.isBulkCheckboxEnabled());
+    workspaces.waitDeleteWorkspaceBtnDisappearance();
+  }
+
+  @Test
+  public void checkSorting() {
+    workspaces.waitPageLoading();
+    workspaces.clickOnRamButton();
+
+    List<Workspaces.WorkspaceListItem> items = workspaces.getVisibleWorkspaces();
+
+    if (items.get(0).getRamAmount() != BLANK_WS_MB) {
+      workspaces.clickOnRamButton();
+      items = workspaces.getVisibleWorkspaces();
+    }
+
+    try {
+      assertEquals(items.get(0).getRamAmount(), BLANK_WS_MB);
+      assertEquals(items.get(1).getRamAmount(), JAVA_WS_MB);
+    } catch (AssertionError ex) {
+      // remove try-catch block after issue has been resolved
+      fail("Known issue https://github.com/eclipse/che/issues/4242");
+    }
+
+    workspaces.clickOnRamButton();
+
+    items = workspaces.getVisibleWorkspaces();
+    try {
+      assertEquals(items.get(0).getRamAmount(), JAVA_WS_MB);
+      assertEquals(items.get(1).getRamAmount(), BLANK_WS_MB);
+    } catch (AssertionError ex) {
+      // remove try-catch block after issue has been resolved
+      fail("Known issue https://github.com/eclipse/che/issues/4242");
+    }
+
+    workspaces.clickOnProjectsButton();
+
+    items = workspaces.getVisibleWorkspaces();
+    assertEquals(items.get(0).getProjectsAmount(), BLANK_WS_PROJECTS_COUNT);
+    assertEquals(items.get(1).getProjectsAmount(), JAVA_WS_PROJECTS_COUNT);
+
+    workspaces.clickOnProjectsButton();
+
+    items = workspaces.getVisibleWorkspaces();
+    assertEquals(items.get(0).getProjectsAmount(), JAVA_WS_PROJECTS_COUNT);
+    assertEquals(items.get(1).getProjectsAmount(), BLANK_WS_PROJECTS_COUNT);
+  }
+
+  @Test
+  public void checkSearchField() throws Exception {
+    int nameLength = expectedBlankItem.getWorkspaceName().length();
+    String sequenceForSearch =
+        expectedBlankItem.getWorkspaceName().substring(nameLength - 5, nameLength);
+
+    workspaces.typeToSearchInput(sequenceForSearch);
+
+    List<Workspaces.WorkspaceListItem> items = workspaces.getVisibleWorkspaces();
+    assertEquals(items.size(), 1);
+    assertEquals(items.get(0).getWorkspaceName(), expectedBlankItem.getWorkspaceName());
+
+    workspaces.typeToSearchInput("");
+
+    items = workspaces.getVisibleWorkspaces();
+    assertEquals(items.size(), 2);
```

Comment:
From experience it's interesting to see whole list items if size of list is incorrect. From this point comparing expected list content with actual one as Strings could bring more information in case of fail, for example https://github.com/eclipse/che/blob/6.6.0/selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/projectexplorer/DownloadProjectTest.java#L152